lib/pavilion/filters/parse_time.py

```diff
@@ -62,12 +63,16 @@ def parse_duration(rval: str, now: datetime) -> datetime:
         raise ValueError(f"Invalid unit {unit} for duration")
 
     if unit == 'years':
-        return now.replace(year=now.year - mag)
+        return safe_update(now, year=now.year - mag)
 
```

Comment:
It feels like using datetime.timedelta would be good enough, but this is better.